### PR TITLE
Remove old function and use elastic search for actor validation

### PR DESCRIPTION
**Pull request by Wal8800**
Key things

- Remove `findMovieByActor`, `findMovieByPlot`, `findMovieByQuote`
- Added some error/validation handling on the lex (might need to fine tune to improve it)
- Added another elastic search function for general query
- clear sessionAttributes when use say they found their movie
- store slot info into sessionAttributes

Todo

- refactor the elastic search query function
- improve lex flow

### CODE REVIEW

File: lambda/dispatcher/SourceDispatcher.js [hunk 1]
```diff
@@ -59,14 +59,15 @@ var dispatch = function(queryInfo) {
...
-    if (queryInfo.slots.hasOwnProperty(slotName)) {
-        return queryInfo.slots[slotName];
-    }
-
     if (queryInfo.sessionAttributes.hasOwnProperty(slotName)) {
         return queryInfo.sessionAttributes[slotName];
     }
 
+    if (queryInfo.slots.hasOwnProperty(slotName)) {
+        queryInfo.sessionAttributes[slotName] = queryInfo.slots[slotName];
```
Comment (changkon) [inline lambda/dispatcher/SourceDispatcher.js:67]: What is this for?

File: lambda/movieFinder.js [hunk 2]
```diff
@@ -11,7 +13,17 @@ var find = function(slots, sessionAttributes) {
...
         .then(() => findMovieFromSources(queryInfo))
         .then((movieLists) => processMovieLists(movieLists))
-        .then((singleMovieList) => resolve(singleMovieList))
+        .then((singleMovieList) => {
+            if (singleMovieList.length == 0 ) {
+                var err = {
+                    type: "NotFound"
+                }
+                
+                reject(err)
```
Comment (changkon) [inline lambda/movieFinder.js]: I was thinking of returning an error object. https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Error
We can add the message inside the error object

File: lambda/moviebotFunction.js [hunk 3]
```diff
@@ -261,13 +130,22 @@ function findMovie(intentRequest, callback) {
         callback(confirmIntent(sessionAttributes, 'ContinueFinding', {}, msg, movieToResponseCards(singleMovieList)))
     }).catch((err) => {
         console.log(err)
-        //error object structure
         if (err.type === "Validation") {
-
+            if (intentName === "FindMovieByActor") {
```
Comment (changkon) [inline lambda/moviebotFunction.js]: Refer again to https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Error

File: lambda/dispatcher/SourceDispatcher.js (hunk 1 above)
Comment (Wal8800) [reply lambda/dispatcher/SourceDispatcher.js:67]: store the slot info into session attributes

**Review by WesleyYep — APPROVED** [review]

**Review by changkon — APPROVED** [review]

**Pull request merged**